inst/private/python_ipc_pytave.m

```diff
@@ -59,17 +61,50 @@
 
   info.prelines = 0;
 
-  % TODO: bit wasteful to repeat this every call
-  headers = python_header();
-  pyexec(headers)
+  if isempty(have_headers)
+    pyexec(strjoin({'import sys',
+                    'import sympy',
+                    'import sympy as sp',
+                    'from sympy import __version__ as spver',
+                    'from sympy import *',
+                    'from sympy.logic.boolalg import Boolean, BooleanFunction',
+                    'from sympy.core.relational import Relational',
+                    '# temporary? for piecewise support',
+                    'from sympy.functions.elementary.piecewise import ExprCondPair',
+                    'from sympy.integrals.risch import NonElementaryIntegral',
+                    'from sympy.matrices.expressions.matexpr import MatrixElement',
+                    '# for hypergeometric',
+                    'from sympy.functions.special.hyper import TupleArg',
+                    'from sympy.utilities.iterables import uniq',
+                    'import copy',
+                    'import binascii',
+                    'import struct',
+                    'import codecs',
+                    'import xml.etree.ElementTree as ET',
+                    'from distutils.version import LooseVersion',
```

Comment:
Just FYI, let me walk you through it:

`git blame python_header.py`

````
---- 8< ----
7b6c35b5 (Colin B. Macdonald 2014-11-16 15:23:32 +0000  35)     from sympy.functions.elementary.piecewise import ExprCondPair
51a80384 (Colin B. Macdonald 2016-01-07 00:26:37 -0800  36)     from sympy.integrals.risch import NonElementaryIntegral
9474bd5a (Colin B. Macdonald 2016-01-11 00:05:02 -0800  37)     from sympy.matrices.expressions.matexpr import MatrixElement
400811d8 (Colin B. Macdonald 2016-05-09 13:04:12 -0700  38)     # for hypergeometric
400811d8 (Colin B. Macdonald 2016-05-09 13:04:12 -0700  39)     from sympy.functions.special.hyper import TupleArg
af6e79e7 (latot              2016-01-17 02:01:21 -0300  40)     from sympy.utilities.iterables import uniq
---- 8< ----
````

So lets look at this one: `from sympy.integrals.risch import NonElementaryIntegral`:

`git show 51a80384`

````
commit 51a80384d5caea0db211e452d39f0b4f6b3778cc
Author: Colin B. Macdonald <cbm@m.fsf.org>
Date:   Thu Jan 7 00:26:37 2016 -0800

    Fix passing NonElementaryIntegrals back to Python
    
    Fixes https://savannah.gnu.org/bugs/index.php?46831.
    
    Yet another case where `eval(srepr(...))` is not quiet the
    identity map.
````

(this doesn't always work b/c the most recent change to a line might not explain why that line is there...)